src/main/java/com/neuvector/Scanner.java

```diff
@@ -329,6 +332,30 @@ private static String maskCredential(String message, String credential){
         return message.replace(credential, "******");
     }
 
+    private static String getDockerUserAndGroup() {
+        String user = executeBashCommand("id -u");
+        String group = executeBashCommand("id -g");
```

Comment:
I do not understand this implementation. What does this do? My understanding from the ticket is, the user is already passing the user they want to run docker with.. Is this guranteed to have the same mapping? Does what the user suggest not work, i.e.

```
try {
    String userName = System.getProperty("user.name");
    String command = "id -u "+userName;
    Process child = Runtime.getRuntime().exec(command);

    // Get the input stream and read from it
    InputStream in = child.getInputStream();
    int c;
    while ((c = in.read()) != -1) {
        process((char)c);
    }
    in.close();
} catch (IOException e) {
}
```

?

Reply:
For this one I am using just the `id- u `command to get the current user and group for the host machine, for example in my case I got `1000` which is the same if I pass` id -u rodolfo`

Reply:
What does id -g do? I think it is not needed?

Reply:
Can you check if the following works?

```
String username = System.getProperty("user.name");
```

and then in the docker command:
```
"-u", username
```

Reply:
I got: 

`docker run -u rodolfo --name ZFDAGL --rm -v /var/run/docker.sock:/var/run/docker.sock -v /temp:/var/neuvector -e SCANNER_REPOSITORY=library/alpine -e SCANNER_TAG=3.6 -e SCANNER_LICENSE=****** -e SCANNER_REGISTRY=https://registry.hub.docker.com -e SCANNER_REGISTRY_USERNAME= -e SCANNER_REGISTRY_PASSWORD= registry.hub.docker.com/neuvector/scanner:latestdocker: Error response from daemon: unable to find user rodolfo: no matching entries in passwd file.`

Reply:
No.. I mean the code should change.. Not when you are running.. In line 59 (and all others), instead of `dockerUserAndGroup` use `System.getProperty("user.name")`, i.e. `userName`..

Reply:
If I run with the user group such as, it works:

`docker run -u 1000 --name JSZPZC --rm -v /var/run/docker.sock:/var/run/docker.sock -v /temp:/var/neuvector -e SCANNER_REPOSITORY=library/alpine -e SCANNER_TAG=3.6 -e SCANNER_LICENSE=r+wzvemF4mHMIBkvKHXTuuLzw4ML/RW2ARfbC06Qem1VdQ8DXcMWNKbd8GD0uYiLoS9d9XPM8bff4ag66fqwdOa42oNXk9dzP1TTRg0zvVKJNMxbxKAuGMY24x0NlXstc3Jl2BcXq7aTHrlQwzJ/7qD8157+YfcjKJbywbt3WvvdpuoJtJIuKaO9hOfGu9/zbRZznEtZiJj56xaRMVZ/ZkPHIMseRWML7FUr1oft20PYU7ltf1BGPT/LlocAXAJucBpUz5iosRzS1iYJhU4M1kdBhEwQHl6FZXrtzJGYpM21587FyYrkvoIPQNlIijaAOiCXR060PJNfcQzOzEpYRidhVPnHIHQuIn8gcYuU3XwPB9Pq3tGzERpr0aB64EjSSanpcIh/XEhnXijiRP6CfxsHx8RvD/jkXy5VlJpTVFh1Iu7ywE6fePHb3hIe16uCtdYB8fu2p8ksCcnxXbjBUkKvdLLxq5gvOCfCJxknd4GdIS3UJySNblgmlfU6liYcgt555TitN5DAjuMhTaoxj9FoPPwJsl4rrf2+32cqUFI+9+UdNsSd2gWtS/8T2ifPHNVMA7fdPdIWx+/4jXP8u484cLqGG9RutNGPo0k7cmxmgx29FvVOMvpkotD8IZsbCfl8sFHtFRP7PegWYPghZf6yUkw5m4SD+bacVZalnLizZKbDgFuzPmH1LgrroVN1Ngf06JVfO6lDodbAOactsJC7P8DOPRUoOtnCf3ffgshPh+QSTKDOmNvFU51VB7MoLlZiKNaVP6yMqS1bUwxRpI5eehqhqCHy46PWfRHF2EnbiQOyYIjJSY5jE54TzOaKZ613GgdIaMN+mbXg+UWajDevwCvKQo+CrwpBoX72tP/as9G3zIJILaKuXbQRvTtZbgGREMSoLIDxPBYe3FVqMHFg/jUE9iPbsF16JHv/z3EzcAYKwYSyDEyVppTvLjQY0t/Vul4SEj+QWpyW7elAUmatzHdAcX2SB3oDOhUtDyIxw8rtenD9hCOfspJtEFknhtM+Zm1WuoQYRM5/paEVvbHZnu2XHY2cnBtxJZAD8wKktnSPGg3foCXQpQcSEyhfWhVT14WX2/GxgbdwIuxSBcoHlcduca5tKtlKAhJ4kjPPZeQc5A18z48pLRU= -e SCANNER_REGISTRY=https://registry.hub.docker.com -e SCANNER_REGISTRY_USERNAME= -e SCANNER_REGISTRY_PASSWORD= registry.hub.docker.com/neuvector/scanner:latest`

Reply:
You do not need the `--name ZFDAGL` btw..